File: python-implementation/drag_torque.py
import numpy as np

def drag_torque(motor, rpm): 
    T_d = np.zeros((3,1))
    T_d[2,0] = np.sum(motor.k_d * np.multiply(rpm,np.multiply((np.absolute(rpm)),motor.rot_dir)))
    return T_d

File: drag_torque.c
#include <stdio.h>
#include <math.h>


void drag_torque(struct Motor SpecificMotor, double rpm[4][1], double T_d[3][1]){


T_d[0][0] = 0;
T_d[1][0] = 0;
double sum = 0;
double totalSum = 0;

for(int i = 0; i < 4; i++){
	for(int j = 0; j < 1; j++){
		sum = SpecificMotor.k_d * rpm[i][j] * fabs(rpm[i][j]) * SpecificMotor.rot_dir[i][j];
	}
	totalSum = totalSum + sum;
	sum = 0;	
}
T_d[2][0] = totalSum;

}




